### ISSUE TITLE
py2 ↔ py3 compatibility

### ISSUE THREAD

**Issue by gferreira** [typemytype/oneToThreeRoboFontExtension#3]
shouldn’t the extension also try to handle [these](http://python-future.org/compatible_idioms.html#essential-syntax-differences)?

**Comment by typemytype** [comment]
guess not, that would just copy that whole module...

**Comment by gferreira** [comment]
ok, let me try again –

there are two levels of API differences between RF1 & RF3:

- RoboFab ↔ FontParts
- py2 ↔ py3

if I understand it correctly, the purpose of this extension is to make RF1 code work in RF3, and RF3 code work in RF1 — *automatically*. in order for this to happen, it would be necessary to handle py2/py3 differences too.

- RF1 supports some py3 syntax, for example `print()` and unicode literals. float division, on the other hand, depends on the preference setting *Use Future Division*.
- RF3 does not support py2 syntax like `print` and `reload`

```python
# print 'hello world' # doesn't work in RF3
print('hello world')

# this works
s1 = 'The Zen of Python'
s2 = 'きたないのよりきれいな方がいい\n'
print(s1)
print(s2)

print(10 / 3) # float division depends on preference
print(10 // 3)

import sys
# reload(sys) # doesn't work in RF3
# from importlib import reload # doesn't work in RF1

# this works in RF1 & RF3
from mojo.roboFont import version
if version >= "3.0.0":
    from importlib import reload
reload(sys)
```

I think the bottom line is:

- it is not possible to convert RF1 code into RF3 code (and vice-versa) automatically
- it is possible to make changes to the code (manually), so it works in both RF1 & RF3

**Comment by gferreira** [comment]
my point is: I’m not sure if converting between RF1 ↔ RF3 syntax automatically is a reasonable goal.

I think we should focus our efforts into documenting the differences between RF1/RF3 APIs (RoboFab/FontParts & py2/py3), so that developers can update their code manually.

hope this makes sense…

**Comment by gferreira** [comment]
see [Upgrading code to RoboFont 3](https://typemytype.gitlab.io/robofont_com/documentation/building-tools/recommendations-upgrading-RF3/)

**Comment by typemytype** [comment]
the main goal is that py2 RF3 code is compatible with py2 RF1

the secondary goals is to make the transition as smooth as possible.... so it has to go in both ways

**Comment by gferreira** [comment]
added text to the README to clarify that this extension does not handle py2/py3 conversion: 3fa4f59
